packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts

```diff
@@ -0,0 +1,60 @@
+import { Compilation } from 'webpack';
+import type { Compiler, sources } from 'webpack';
+
+/**
+ * Forces all files with `griffel.css` be concatenated into a single asset.
+ */
+function forceCSSIntoOneStyleSheet(compiler: Compiler) {
+  compiler.options.optimization = compiler.options.optimization || {};
+  compiler.options.optimization.splitChunks = compiler.options.optimization.splitChunks || {};
+  compiler.options.optimization.splitChunks.cacheGroups = compiler.options.optimization.splitChunks.cacheGroups || {};
+
+  compiler.options.optimization.splitChunks.cacheGroups['griffel'] = {
+    name: `griffel`,
+    type: 'css/mini-extract',
+    chunks: 'all',
+    test: /griffel.css/,
+    enforce: true,
+  };
+}
+
+function getAssetSourceContents(assetSource: sources.Source): string {
+  const source = assetSource.source();
+
+  if (typeof source === 'string') {
+    return source;
+  }
+
+  return source.toString();
+}
+
+export class GriffelCSSExtractionPlugin {
+  static loader = require.resolve('./webpackLoader');
+
+  apply(compiler: Compiler): void {
+    forceCSSIntoOneStyleSheet(compiler);
+
+    compiler.hooks.compilation.tap('GriffelExtractPlugin', compilation => {
+      compilation.hooks.processAssets.tap(
```

Comment:
you only tap into `processAssets` after `compilation` to make sure that the build time transforms have all finished running ?

Reply:
I forgot to drop this comment... The whole part with `compiler.hooks.compilation.tap` is currently useless as the sorting/deduplication process is not implemented yet. Technically it can be deleted from the current PR, but I will need this part in the next PR anyway.

Reply:
why would you need to tap into the compilation hook ? the sorting/deduplication will still happen on asset processing level

Reply:
Because I need to have an access to `compilation` to replace assets?

```
   compiler.hooks.compilation.tap('GriffelExtractPlugin', compilation => {
      compilation.hooks.processAssets.tap(
        {
          name: 'GriffelExtractPlugin',
          stage: Compilation.PROCESS_ASSETS_STAGE_PRE_PROCESS,
        },
        assets => {
```

- [compilation](https://webpack.js.org/api/compiler-hooks/#compilation) is a _compiler_ hook
- [processAssets](https://webpack.js.org/api/compilation-hooks/#processassets) is a _compilation_ hook

Reply:
aaaah ok, I missed that the compilation object is provided by the compilation hook